saleor/graphql/giftcard/filters.py

```diff
@@ -128,14 +130,31 @@ def filter_events_by_type(events: List[models.GiftCardEvent], type_value: str):
 
 
 def filter_events_by_orders(events: List[models.GiftCardEvent], order_ids: List[str]):
-    _, order_pks = resolve_global_ids_to_primary_keys(order_ids, "Order")
+    order_pks = _get_order_pks(order_ids)
+
     filtered_events = []
     for event in events:
-        if str(event.parameters.get("order_id")) in order_pks:
+        if event.order_id in order_pks:
             filtered_events.append(event)
     return filtered_events
```

Comment:
Maybe we could rewrite it to don't use `for` loop.  We can use `Exists` on `OrderQS` with `Q(id__in=pks) | (Q(use_old_id=True) & Q(number__in=old_pks))`  lookup.

Reply:
In this case, we will add an additional query, as the `events` are in the list.

Reply:
Yup. It's my mistake. My assumption is that `events` is QS. Let's leave it as it is now.